### ISSUE TITLE
Revise FT Scholastic Standing Restriction Logic - automatic trigger (part 2 FT SS)

### ISSUE THREAD

**Issue by HRAGANBC** [bcgov/SIMS#2197]
Revise FT Scholastic Standing Restriction Logic: build automatic trigger.

**Acceptance Criteria**
Build an automatic trigger when below conditions are met (system action trigger ONLY):
- [x] If student has a RESOLVED SSR restriction on their account and ANY positive number of unsuccessful weeks is reported on a student Full-Time application, then a new SSRN restriction must be applied. 
- [x] If student has a RESOLVED SSR restriction on their account and new WTHD reported on a student Full-Time application, then a new SSRN restriction must be applied. 
- [x] If student has a RESOLVED SSRN restriction on their account and ANY positive number of unsuccessful weeks is reported on a student Full-Time application or new WTHD reported on a student Full-Time application, then a new SSRN restriction must be applied. 
- [x] Unsuccessful weeks are reported for student FT application, and if the cumulation of unsuccessful weeks across all FT applications >=68, then SSR is applied: FT UC counter exceeds or equal 68 (>=). 

<img width="718" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeFVQQ2c9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c408254a857882f82f29c96504aac748ee8be96e/image.png" alt="image.png" />

- [x] If student already has an ACTIVE SSR, do not trigger second SSR. 
- [x] If student already has an ACTIVE SSRN, do not trigger second SSRN. 

## Technical Details:
https://github.com/bcgov/SIMS/blob/main/sources/packages/backend/apps/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts#L307-L360

## Test scenarios for automatic SSRN and SSR:
1. Manually add WTHD restriction on student profile as a Ministry. Submit FT application as Student, accept NOA. Confirm COE as Institution. Report Withdraw as Institution -- >  SSR is automatically triggered. 
2.  - Manually add SSR as a Ministry User. Resolve SSR as a Ministry user. Submit FT application as Student, accept NOA. Confirm COE as Institution. Report Withdraw as Institution -- > WTHD is automatically added by system and SSRN is automatically triggered. 
- Manually add SSR as a Ministry User. Resolve SSR as a Ministry user. Submit FT application as Student, accept NOA. Confirm COE as Institution. Report Unsuccessful Completion as Institution -- > FT UC counter is updated and SSRN is automatically triggered. 

3. Student has no prior restrictions. Submit FT application as Student, accept NOA. Confirm COE as Institution. Report Unsuccessful Completion as Institution. Submit second FT application as Student, accept NOA. Confirm COE as Institution. Report  second UC as Institution. Make sure combined number of reported UC weeks is equal or greater than 68 weeks --- > FT UC counter is updated. Once FT UC counter is equal or greater than 68 weeks,  SSR is automatically triggered. 

_In the following test cases SSR should not be triggered:_ 

1. - Manually add SSR as a Ministry User. Resolve SSR as a Ministry user. Submit PT application as Student, accept NOA. Confirm COE as Institution. Report Withdrawal as Institution -- > PTWTHD is automatically added by system but SSRN should NOT be triggered. 
- Manually add SSR as a Ministry User. Resolve SSR as a Ministry user. Submit PT application as Student, accept NOA. Confirm COE as Institution. Report Unsuccessful Completion as Institution -- >  SSRN should NOT be triggered. 

2. Please test the bulk upload_ Scholastic Standing

**Technical ACs**

- [ ] When scholastic standing type `Student did not complete program`  is reported.
  - [ ] If `SSR` is present in the student account (active or inactive), create an `SSRN`.
  - [ ] If `SSRN` is already present and active, do not add it.
  - [ ] If `SSR` and `SSRN` were never added to the student account
    - [ ] Check if the total number of unsuccessful weeks is greater than or equal to 68
      - [ ] If yes, create `SSR`,
      - [ ] If not, do nothing.
- [ ] When scholastic standing type `Student withdrew from program` is reported.
  - [ ] If `SSR` is present in the student account (active or inactive), create an `SSRN`.
  - [ ] If `SSRN` is already present and active, do not add it.
  - [ ] If an active `WTHD` is not present, add a `WTHD` .
  - [ ] If an active `WTHD` is present, add an `SSR`.

**Comment by AnnaPBashkatova** [comment]
Pulling back into SOAK, as we learned that new SSRN restriction is expected to be built. Will be adding SSRN to the ticket. Points will need to be reevaluated.

**Comment by AnnaPBashkatova** [comment]
@Joshua-Lakusta 
Hi Josh, we added the AC here about resolved SSRN. Please read the third AC on the ticket and let me know if you want to change anything here.

**Comment by andrewsignori-aot** [comment]
@AnnaPBashkatova, the below AC does not state that SFAS total weeks should be considered. Is this a gap or something that will be addressed in another ticket?

> Unsuccessful weeks are reported for student FT application, and if the cumulation of unsuccessful weeks across all FT applications >=68, then SSR is applied: FT UC counter exceeds or equal 68 (>=).

**Comment by AnnaPBashkatova** [comment]
@andrewsignori-aot Good catch, Signori. I will address it another ticket, if that works for you.

**Comment by andrewsignori-aot** [comment]
@Joshua-Lakusta @AnnaPBashkatova to ensure that we are on the same page, related to the bulk withdraw mentioned in the ticket test scenarios, only UI was implemented in the past during the #2111, which means, no file processing is actually happening right now, other than the file validation.

**Comment by AnnaPBashkatova** [comment]
@andrewsignori-aot  Good morning, Andrew. "Assuming the student has an SSR or even an SSRN in his account (resolved), if the student has a new Student did not complete program" the system should generate only SSRN.

**Comment by andrewsignori-aot** [comment]
@AnnaPBashkatova regarding the AC that reads

> If student has a RESOLVED SSR restriction on their account and ANY positive number of unsuccessful weeks is reported on a student Full-Time application, then a new SSRN restriction must be applied.

Is SIMS supposed to allow a UC scholastic standing to be reported with 0 weeks?

**Comment by AnnaPBashkatova** [comment]
@andrewsignori-aot  As we discussed, will create a new ticket to change the validation on UC reporting.

**Comment by andrewsignori-aot** [comment]
## Demo

BCeID Student: andrew_signori_ssr2
Institution: SIMS_COLLE

Created three student applications
- 2025000091 used for `Student withdrew from funded period` first time.
- 2025000092 used for `Student withdrew from funded period` second time.
- 2024000193 used for `Student withdrew from funded period`
  - 2024-25
  - Program "Bulk Offering Upload Parallel Refactor Demo", Offering 500

- Use the brand new student account with no restrictions
- In the institution portal, execute the UC for 2025000091 reporting 70 weeks of UC.
  - SSR should be created in the student account.
- In the institution portal, execute the UC for 2025000092 reporting 1 week UC
  -  SSRN should be created in the student account.
-  Resolve both restrictions in the student account
-  In the institution portal, execute the `Student withdrew from funded period`.
  -  WTHD should be added
  -  SSRN should be added
